### DISCUSSION TITLE
how can i use oak with sono

### DISCUSSION THREAD

**Discussion by monotter** [oakserver/oak#461] [Q&A]
i made below code to make sono connection with oak
```ts
import { SONO , OAK } from './Dependencies.ts'
const { Application } = OAK
const { Sono } = SONO
const app = new Application()
const sono = new Sono()
app.listen({ port: parseInt(Deno.env.get('PORT')!) })
app.use((context, next) => {
    if (context.request.method == 'GET' && context.request.url.pathname == '/sono') {
        sono.connect(context.request.originalRequest, () => {
            sono.emit('new client connected')
        })
    } else { next() }
})
```
`./Dependencies.ts`
```ts
import * as SONO from 'https://deno.land/x/sono@v1.1/mod.ts'
import * as OAK from 'https://deno.land/x/oak@v10.1.1/mod.ts'
export { SONO, OAK }
```
but it wont work because oak's `context.request.originalRequest` does not includes `bufWriter` and `bufReader` but deno's ServerRequest have them

**Comment by kitsonk** [comment]
sono uses the deprecated Deno `std` HTTP server and WebSocket integration.  Current versions of oak use the Deno native implementations which are compatible with both the Deno CLI and Deploy.

So basically the packages are incompatible, and you would need to ask the developers of sono to update their library to utilise the native HTTP server and WebSocket APIs in Deno.
